File: src/main/kotlin/com/example/webshop/security/ResourceServer.java
package com.example.webshop.security;

import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;

@EnableResourceServer
public class ResourceServer extends WebSecurityConfigurerAdapter{

    /*
    TODO Zapezpieczyć endpointy
    /oauth/token musi być publiczny
     */
    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeRequests()
                .antMatchers("*")
                .permitAll();
    }
}

File: src/main/kotlin/com/example/webshop/security/ResourceServer.kt
package com.example.webshop.security

import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer
import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter


@EnableResourceServer
@Configuration
@EnableGlobalMethodSecurity(prePostEnabled = true)
@EnableWebSecurity
class ResourceServer : ResourceServerConfigurerAdapter() {

    /*
    TODO Zapezpieczyć endpointy
    /oauth/token musi być publiczny

     */

    override fun configure(http: HttpSecurity) {
        http
                .authorizeRequests()
                //TODO zabezpieczyć endpointy
                .antMatchers("/oauth/token").permitAll()
                .antMatchers("/user","/users/*/isActivated", "/user/activate/*", "/user/shop", "/user/vendor", "/shops").permitAll()
                .antMatchers(
                        "/products",
                        "/products/shop/*", "/products/*",
                        "/products/category/*",
                        "/products/").permitAll()
                .anyRequest().authenticated()
    }
}
